### PR TITLE
[mongodb]: Allow all objects for pull

### PR DESCRIPTION
**Pull request by floric**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.mongodb.com/manual/reference/operator/update/pull/
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by floric** [comment]
fixes #29911 for now, but could be better in the future.

**Review by daprahamian — APPROVED** [review]

**Review by CaselIT — APPROVED** [review]

File: types/mongodb/index.d.ts [hunk 1]
```diff
@@ -869,7 +869,7 @@ export type UpdateQuery<T> = {
     $currentDate?: { [P in keyof T]?: (true | { $type: 'date' | 'timestamp' }) } | { [key: string]: (true | { $type: 'date' | 'timestamp' }) };
     $addToSet?: Partial<T> | { [key: string]: any };
     $pop?: { [P in keyof T]?: -1 | 1 } | { [key: string]: -1 | 1 };
-    $pull?: Partial<T> | { [key: string]: Condition<T, keyof T> };
+    $pull?: Partial<T> | { [key: string]: Condition<T, keyof T> | any };
```
Comment (rbuckton) [inline types/mongodb/index.d.ts]: This weakens `$pull` to `Partial<T> | { [key: string]: any }` as the union of `Condition<T, keyof T> | any` is reduced to `any`.

If you intend to weaken the type to `any`, I would suggest you remove `Condition<T, keyof T>` from the union.

**Review by rbuckton — CHANGES_REQUESTED** [review]

File: types/mongodb/index.d.ts (hunk 1 above)
Comment (DanielRosenwasser) [reply types/mongodb/index.d.ts]: Also, can you provide a test case that you need working in the tests file?

**Comment by floric** [comment]
I'm still not really happy about the change. Do you know why `$where` is mandatory? Like this, it gets suggested in the IDE.

**Pull request closed without merging**